flang/lib/Lower/OpenMP/ClauseProcessor.cpp

```diff
@@ -919,8 +924,27 @@ bool ClauseProcessor::processMap(
 
         for (const Fortran::parser::OmpObject &ompObject :
              std::get<Fortran::parser::OmpObjectList>(mapClause->v.t).v) {
+          llvm::omp::OpenMPOffloadMappingFlags objectsMapTypeBits = mapTypeBits;
+          checkAndApplyDeclTargetMapFlags(converter, objectsMapTypeBits,
+                                          *getOmpObjectSymbol(ompObject));
+
           llvm::SmallVector<mlir::Value> bounds;
           std::stringstream asFortran;
+          const Fortran::semantics::Symbol *parentSym = nullptr;
+
+          if (getOmpObjectSymbol(ompObject)->owner().IsDerivedType()) {
+            const auto *designator =
+                Fortran::parser::Unwrap<Fortran::parser::Designator>(
+                    ompObject.u);
+            assert(designator && "Expected a designator from derived type "
+                                 "component during map clause processing");
+            parentSym = GetFirstName(*designator).symbol;
```

Comment:
We will need to traverse all parents.  For example given `x%y%z`, `GetFirstName` will return `x`.  The type of `x` does not have a member `z`, so findComponentMemberPlacement will fail.

Reply:
This PR isn't intended to handle that type of mapping yet, it'll be handled in a subsequent series (which I am working on at the moment). 

It's only intended to handle cases where we're mapping one level of nesting for the moment.